build.sbt

```diff
@@ -1,31 +1,62 @@
-crossScalaVersions := Seq("2.13.6", "2.12.14")
-scalaVersion := crossScalaVersions.value.head
+import xerial.sbt.Sonatype._
 
-name := "library-example"
+ThisBuild / scalaVersion := "3.0.1-RC2"
```

Comment:
Should we set the Scala version directly in the `documentation` project instead of setting it globally?

Reply:
I thought that most projects will share the same `scalaVersion` across all modules and this avoids repeating the declaration

Reply:
This is not the case here, `documentation` is the only module that uses that Scala version.

Reply:
I am a bit confused. `documentation/doc` runs the `doc` task of `library`. So maybe no module is using that version?

Reply:
It depends how you invoke `documentation/doc`. I’m not sure what happens if you run `+documentation/doc`, or `++ 3.0.1-RC2 documentation/doc`. Sbt’s behavior in these cases is weird.

But I thought your intent was to use a different version of Scala for `library` and `doc` (otherwise you wouldn’t redefine `scalaVersion`/`crossScalaVersions` within `library`?).

Reply:
It was my intent but I am not so sure anymore that it is possible

Reply:
So, if your intent is to use a different version of Scala for `library` and `site`, then you should really set the setting `scalaVersion` within the scope of each respective project, not globally.

Reply:
My theory is that if I set `scalaVersion := x` in `site`, then `x` will be used for mdoc but not for `library/doc` which will continue to use its own version of scaladoc to generate the site and lib api

Reply:
Unfortunately, the behavior of sbt here depends on how we invoke the task `doc`. If we write `++ 3.0.0 site/doc`, it sets the Scala version for both `library` and `site`. If we just invoke `doc`, then it depends on the default version set per project (so, here it’s `3.0.0` for `library` and `3.0.1-RC2` for `doc`).

Reply:
I agree. So I think we should set a global `scalaVersion` and remove the repetition in each module.
We can keep `crossScalaVersions` but I do not think that scaladoc is able to run for them

Reply:
OK, what about the following?

~~~ scala
val scala3Version = "3.0.1"
ThisBuild / scalaVersion := scala3Version

val library = project
  .settings(
    crossScalaVersions := List(scala3Version, "2.13.6")
  )

val site = project
  .settings(
  )
~~~